### PR TITLE
Add Neo4j::Types v2 support to feat-v5.0 branch

### PR DESCRIPTION
**Pull request by johannessen**
This PR is supposed to make the [feat-v5.0 branch](https://github.com/majensen/perlbolt/tree/feat-v5.0) compliant with the [proposed Neo4j::Types v2](https://github.com/johannessen/neo4j-types/milestone/1) interface.

The methods for accessing the element ID are complete.

*(See ["element_id" in N:T:Node](https://github.com/johannessen/neo4j-types/blob/37b415c33a7d6055a604f853bca88f352d8d8795/lib/Neo4j/Types/Node.pod#methods) and ["Node" in N:T:ImplementorNotes](https://github.com/johannessen/neo4j-types/blob/37b415c33a7d6055a604f853bca88f352d8d8795/lib/Neo4j/Types/ImplementorNotes.pod#node) for the specification, and [neo4j-types #13 + #3](https://github.com/johannessen/neo4j-types/pull/13) for the rationale. Even though I have now made a decision for Neo4j::Types, of course I will always keep the door open to revisit it. If you disagree with me, I’d love to consider your thoughts.)*

The availability check currently uses `eq`, [as discussed earlier](https://github.com/majensen/perlbolt/issues/50#issuecomment-1596162775). A simpler way to avoid `eq` might be to move this check to XS and look at the PV/IV flags instead. But this is just about performance – the code works fine as it is.

### CODE REVIEW

**Comment by johannessen** [comment]
The methods for Point, DateTime, and Duration are complete as well. However, I see two problems that still remain to be solved:

* [DateTime](https://github.com/johannessen/neo4j-types/blob/cff7afd2298809b2c30e8f8e1ce1f81d3561e639/lib/Neo4j/Types/DateTime.pod) and [Duration](https://github.com/johannessen/neo4j-types/blob/cff7afd2298809b2c30e8f8e1ce1f81d3561e639/lib/Neo4j/Types/Duration.pod) are not released on CPAN yet. They will be part of Neo4j::Types v2. To get the code to compile and make the tests pass anyway, the modules here in perlbolt include these two as private packages, which is a bit of an ugly hack that should be removed once Neo4j::Types v2 is released.

* There is no capability yet in CTypeHandlers.xs for accepting generic Neo4j::Types values and converting them into the `neo4j_*` types. Since Point, DateTime, and Duration may be used as query parameters, this needs to be added to [SV_to_neo4j_value()](https://github.com/majensen/perlbolt/blob/v0.5000rc/lib/Neo4j/Bolt/CTypeHandlers.xs#L110).

    Basically, if the given SV is blessed, we need to call the `DOES()` method to see if the SV performs one of these roles (instead of `sv_isa()`); if so, we need to its methods to determine the value (instead of `hv_fetchs()` etc.). We could keep the current code around as a small optimisation for Neo4j::Bolt types, or alternatively we could remove the current code to simplify future maintenance.

**Comment by johannessen** [comment]
Bytes, DateTime, Duration, and Point are complete and ready to be merged into `feat-v5.0`.
